src/charm.py

```diff
@@ -659,6 +660,10 @@ def _on_nrpe_targets_changed(self, event: Optional[NrpeTargetsChangedEvent]):
                 nrpe["target"], nrpe["app_name"], **nrpe["additional_fields"]
             )
 
+        # Add vector scrape job
+        target = {self.unit.name: {"hostname": self.host, "port": VECTOR_PORT}}
+        self.metrics_aggregator.set_target_job_data(target, self.app.name)
```

Comment:
How does this work? Are we not losing the nrpe scrape jobs by calling this function?

Reply:
We call here this public method, the same way we do [some line above to add the nrpe targets](https://github.com/canonical/cos-proxy-operator/blob/scrape_vector/src/charm.py#L659), look:

```python
        for nrpe in nrpes:
            self.metrics_aggregator.set_target_job_data(
                nrpe["target"], nrpe["app_name"], **nrpe["additional_fields"]
            )
```

We are not losing nrpe scrape jobs.